### ISSUE TITLE
pip install "satpy[all]" fails

### ISSUE THREAD

**Issue by raybellwaves** [pytroll/satpy#1347]
**Describe the bug**
Followed the installation instructions in the docs (https://satpy.readthedocs.io/en/latest/install.html). Note: made a PR as `pip install satpy[all]` fails
```
(test_env) Ray@UM-404XFVH4 ~ % pip install satpy[all]
zsh: no matches found: satpy[all]
```

**To Reproduce**

```
$ conda create -n test_env python=3.8
$ conda activate test_env
$ pip install "satpy[all]"
```

**Expected behavior**
Install everything/

**Actual results**

```
Collecting satpy[all]
  Downloading satpy-0.22.0.tar.gz (827 kB)
     |████████████████████████████████| 827 kB 1.2 MB/s 
Collecting numpy>=1.13
  Using cached numpy-1.19.1-cp38-cp38-macosx_10_9_x86_64.whl (15.3 MB)
Collecting pillow
  Downloading Pillow-7.2.0-cp38-cp38-macosx_10_10_x86_64.whl (2.2 MB)
     |████████████████████████████████| 2.2 MB 3.3 MB/s 
Collecting pyresample>=1.11.0
  Downloading pyresample-1.16.0-cp38-cp38-macosx_10_9_x86_64.whl (509 kB)
     |████████████████████████████████| 509 kB 3.8 MB/s 
Collecting trollsift
  Downloading trollsift-0.3.4-py2.py3-none-any.whl (26 kB)
Collecting trollimage>1.10.1
  Downloading trollimage-1.13.0-py2.py3-none-any.whl (53 kB)
     |████████████████████████████████| 53 kB 2.5 MB/s 
Collecting pykdtree
  Downloading pykdtree-1.3.1.tar.gz (78 kB)
     |████████████████████████████████| 78 kB 3.2 MB/s 
Processing ./Library/Caches/pip/wheels/13/90/db/290ab3a34f2ef0b5a0f89235dc2d40fea83e77de84ed2dc05c/PyYAML-5.3.1-cp38-cp38-macosx_10_9_x86_64.whl
Collecting xarray!=0.13.0,>=0.10.1
  Downloading xarray-0.16.0-py3-none-any.whl (704 kB)
     |████████████████████████████████| 704 kB 2.9 MB/s 
Collecting dask[array]>=0.17.1
  Downloading dask-2.25.0-py3-none-any.whl (834 kB)
     |████████████████████████████████| 834 kB 3.2 MB/s 
Collecting pyproj
  Downloading pyproj-2.6.1.post1-cp38-cp38-macosx_10_9_x86_64.whl (13.0 MB)
     |████████████████████████████████| 13.0 MB 3.3 MB/s 
Collecting zarr
  Downloading zarr-2.4.0.tar.gz (3.3 MB)
     |████████████████████████████████| 3.3 MB 2.5 MB/s 
Collecting pyhdf
  Downloading pyhdf-0.10.2.tar.gz (149 kB)
     |████████████████████████████████| 149 kB 2.8 MB/s 
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Collecting gdal
  Downloading GDAL-3.1.2.tar.gz (593 kB)
     |████████████████████████████████| 593 kB 3.0 MB/s 
Collecting libtiff
  Downloading libtiff-0.4.2.tar.gz (129 kB)
     |████████████████████████████████| 129 kB 3.4 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /Users/Ray/miniconda3/envs/test_env/bin/python3.8 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/19/wvb4n9js73z6j4sv545bfvxm0000gq/T/pip-install-8hx6tmn1/libtiff/setup.py'"'"'; __file__='"'"'/private/var/folders/19/wvb4n9js73z6j4sv545bfvxm0000gq/T/pip-install-8hx6tmn1/libtiff/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/19/wvb4n9js73z6j4sv545bfvxm0000gq/T/pip-pip-egg-info-nmszefsy
         cwd: /private/var/folders/19/wvb4n9js73z6j4sv545bfvxm0000gq/T/pip-install-8hx6tmn1/libtiff/
    Complete output (5 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/19/wvb4n9js73z6j4sv545bfvxm0000gq/T/pip-install-8hx6tmn1/libtiff/setup.py", line 77, in <module>
        from numpy.distutils.core import setup, Extension
    ModuleNotFoundError: No module named 'numpy'
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Environment Info:**
 - OS: OSX

**Comment by djhoese** [comment]
The main issue/issues here are dependency packages not having a wheel package for python 3.8. It is one of the same issues for pykdtree that you've discovered too. I see you are using zsh. Good to know that is what is causing the weird issues when not using quotes.

**Comment by djhoese** [comment]
I could be wrong, but I think the main issue for this has been resolved. That is, libtiff and pykdtree have had wheels published for modern versions of Python (not 3.12 yet). Closing this.
